dask/layers.py

```diff
@@ -961,3 +961,170 @@ def __repr__(self):
         return "DataFrameIOLayer<name='{}', n_parts={}, columns={}>".format(
             self.name, len(self.inputs), self.columns
         )
+
+
+class DataFrameBinOp(Blockwise, DataFrameLayer):
```

Comment:
Should dataframe things be in dask/dataframe/ ?

Reply:
In general, we cannot have any Layer difinitions in an array or dataframe submodule, otherwise the scheduler would be required to import things like numpy and pandas.